### PR TITLE
Enable sdpa for nonzero dropout

### PR DESCRIPTION
**Pull request by drisspg**
PyTorch supports dropout for FlashAttention in SDPA via this PR: https://github.com/pytorch/pytorch/pull/92917
As well the Math fallback supports dropout as well. On the above PR I also posted some speed numbers of the train script. 
On an single A100 machine and using the shakespeare dataset.

Prior to this PR speeds and feeds:
Dropout: 0.0, w/ torch.compile: 3273.17ms per loop
Dropout: 0.1, w/torch.compile: 4731.11ms per loop.

As of current master branch for PyTorch and this PR:
Dropout: 0.0, w/ torch.compile: 3273.17ms per loop
Dropout: 0.1, w/torch.compile: 3774.39ms per loop. 

The reason for the speed difference is the added need for some dynamo graph_breaks. There is a discussion for the reasoning on the PyTorch pull-request. However still a net win.

### CODE REVIEW

**Comment by karpathy** [comment]
ty
